app/components/primer/alpha/action_list/item.rb

```diff
@@ -0,0 +1,230 @@
+          icon: Primer::OcticonComponent,
+          label: Primer::LabelComponent,
+          counter: Primer::CounterComponent
+          # text: ->(text) { text }
+        }
+
+        renders_one :private_trailing_action, types: {
+          icon: Primer::OcticonComponent,
+          svg: lambda { |**system_arguments|
+            # Primer::BaseComponent.new(tag: :svg, **system_arguments)
+          }
+        }
+
+        renders_one :trailing_action, lambda { |**system_arguments|
+          Primer::Beta::IconButton.new(scheme: :invisible, classes: ["ActionListItem-trailingAction"], **system_arguments)
+        }
+
+        renders_many :items, lambda { |**system_arguments|
+          @list.build_item(**system_arguments, root: @root || self).tap do |item|
+            @list.will_add_item(item)
+
+            if item.active? && !@root
+              @content_arguments[:classes] = class_names(
+                @content_arguments[:classes],
+                "ActionListContent--hasActiveSubItem"
+              )
+            end
+          end
+        }
+
+        attr_reader :root, :active, :disabled, :checked, :expanded
+
+        alias active? active
+        alias disabled? disabled
+        alias checked? checked
+        alias expanded? expanded
+
+        def initialize(
+          list:,
+          root:,
+          label:,
+          truncate_label: false,
+          href: nil,
+          role: :listitem,
+          size: DEFAULT_SIZE,
+          scheme: DEFAULT_SCHEME,
+          disabled: false,
+          description_scheme: DEFAULT_DESCRIPTION_SCHEME,
+          select_mode: DEFAULT_SELECT_MODE,
+          checked: false,
+          active: false,
+          on_click: nil,
+          expanded: false,
+          trailing_action_on_hover: false,
+          **system_arguments
+        )
+          @list = list
+          @root = root
+          @label = label
+          @href = href
+          @truncate_label = truncate_label
+          @disabled = disabled
+          @checked = checked
+          @active = active
+          @expanded = expanded
+          @trailing_action_on_hover = trailing_action_on_hover
+          @system_arguments = system_arguments
+
+          @size = fetch_or_fallback(SIZE_OPTIONS, size, DEFAULT_SIZE)
+          @scheme = fetch_or_fallback(SCHEME_OPTIONS, scheme, DEFAULT_SCHEME)
+          @select_mode = fetch_or_fallback(SELECT_MODE_OPTIONS, select_mode, DEFAULT_SELECT_MODE)
+          @description_scheme = fetch_or_fallback(
+            DESCRIPTION_SCHEME_OPTIONS, description_scheme, DEFAULT_DESCRIPTION_SCHEME
+          )
+
+          @system_arguments[:classes] = class_names(
+            @system_arguments[:classes],
+            SCHEME_MAPPINGS[@scheme],
+            "ActionListItem",
+            # "ActionListItem--withActions",
+            # "ActionListItem--trailingActionHover",
+            # "ActionListItem--withActions" if trailing_action.present? this isn't working,
```

Comment:
@joelhawksley I couldn't get these conditional classnames to work, can you take a look?